services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -107,21 +106,24 @@ requestDevice (zusr ::: tid ::: uid ::: _) = do
     userLHStatus <- LegalHoldData.getUserLegalHoldStatus uid
     case userLHStatus of
         UserLegalHoldEnabled -> throwM userLegalHoldAlreadyEnabled
-        UserLegalHoldPending -> requestDeviceFromService
-        UserLegalHoldDisabled -> requestDeviceFromService
+        UserLegalHoldPending -> provisionLHDevice
+        UserLegalHoldDisabled -> provisionLHDevice
   where
-    requestDeviceFromService :: Galley Response
+    provisionLHDevice :: Galley Response
+    provisionLHDevice = do
+        (lastPrekey', prekeys) <- requestDeviceFromService
+        -- We don't distinguish the last key here; brig will do so when the device is added
+        LegalHoldData.insertPendingPrekeys uid (unpackLastPrekey lastPrekey' : prekeys)
+        LegalHoldData.setUserLegalHoldStatus uid UserLegalHoldPending
+        notifyClientsAboutLegalHoldRequest zusr uid lastPrekey' prekeys
+        pure $ responseLBS status204 [] mempty
+
+    requestDeviceFromService :: Galley (LastPrekey, [Prekey])
     requestDeviceFromService = do
         LegalHoldData.dropPendingPrekeys uid
         lhDevice <- LHService.requestNewDevice tid uid
         let NewLegalHoldClient
               { newLegalHoldClientPrekeys = prekeys
               , newLegalHoldClientLastKey = lastKey
               } = lhDevice
```

Comment:
we've talked about this: it's ok if we keep this if you disagree with me, but i'm not a fan of using records to keep the compiler from warning us about changes in types that we pattern match against.  i prefer this:

```
    let NewLegalHoldClient prekeys lastkey = lhDevice
```

it's also less verbose and easier to read, i think.